Quadriga/src/main/java/edu/asu/spring/quadriga/web/config/SimpleSignInAdapter.java

```diff
@@ -0,0 +1,125 @@
+import edu.asu.spring.quadriga.domain.IUser;
+import edu.asu.spring.quadriga.exceptions.QuadrigaNotificationException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.exceptions.UsernameExistsException;
+import edu.asu.spring.quadriga.service.IUserManager;
+import edu.asu.spring.quadriga.service.QuadrigaUserDetails;
+import edu.asu.spring.quadriga.web.login.QuadrigaGrantedAuthority;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+import edu.asu.spring.quadriga.web.manageusers.beans.AccountRequest;
+
+/**
+ * 
+ * This class authorizes (assigns appropriate roles) users authenticated by
+ * Github.
+ * 
+ * @author Chiraag Subramanian
+ *
+ */
+
+public final class SimpleSignInAdapter implements SignInAdapter {
+
+    private final Logger logger = LoggerFactory.getLogger(SignInAdapter.class);
+
+    private IUserManager userManager;
+    private IUserHelper userHelper;
+
+    public SimpleSignInAdapter(IUserManager userManager, IUserHelper userHelper) {
+        this.userManager = userManager;
+        this.userHelper = userHelper;
+    }
+
+    /**
+     * This method checks if Quadriga contains details of the user authenticated
+     * by Github. If the user details are present (which indicates the user is
+     * an authorized user), appropriate roles are assigned, Else if the user
+     * details are not present (which indicates the user is an unregistered /
+     * not approved user, restricted access is assigned)
+     */
+
+    @Override
+    public String signIn(String userId, Connection<?> connection, NativeWebRequest request) {
+        List<GrantedAuthority> authorities = new ArrayList<>();
+        String messageType = null;
+        IUser user = (IUser) userManager.findUserByProviderUserId(connection.getKey().getProviderUserId(),
+                connection.getKey().getProviderId());
+        // if user details is not present in the database, create a database
+        // record for the user.
+        if (user == null) {
+            user = userHelper.createUser(connection);
+            AccountRequest accountRequest = new AccountRequest();
+            accountRequest.setUsername(user.getUserName());
+            accountRequest.setName(user.getName());
+            accountRequest.setEmail(user.getEmail());
+            accountRequest.setSocialSignIn(true);
+            accountRequest.setProvider(user.getProvider());
+            accountRequest.setUserIdOfProvider(user.getUserIdOfProvider());
+            messageType = "1";
+            try {
+                userManager.addNewUser(accountRequest);
+            } catch (QuadrigaStorageException e) {
+                logger.error("Could not add user.", e);
+                messageType = "2";
+            } catch (UsernameExistsException e) {
+                logger.error("Username already in use or user account needs to be approved by the admin.", e);
+                messageType = "3";
```

Comment:
strings are not very good as flags. I would create an enum with a string property and use the enum (here and in the login controller).